lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```
@@ -241,18 +243,41 @@ def _on_peer_cluster_relation_departed(self, event: RelationDepartedEvent) -> No
         if not (trigger_app := cluster_fleet_apps_rels.get(str(event.relation.id))):
             return
 
-        # set the planned units to 0 given we're losing visibility over it from here on
         trigger_app = PeerClusterApp.from_dict(trigger_app)
-        trigger_app.planned_units = 0
-
         self._put_fleet_apps(
             deployment_desc=self.charm.opensearch_peer_cm.deployment_desc(),
             target_relation_ids=target_relation_ids,
             p_cluster_app=trigger_app,
             trigger_rel_id=event.relation.id,
         )
 
-    def refresh_relation_data(
+        # if the trigger app is the failover orchestrator and there are no planned units, delete it
+        if (
+            relation_departure_reason(
+                self.charm, self.relation_name, event.app.name if event.app else ""
+            )
+            == RelDepartureReason.SCALE_DOWN
+        ):
```

Usually the app should be set. If this is for typing errors you should raise instead of passing an empty string. By passing an empty string we are effectively setting up the function for undefined behaviour.

---

You're right, I suppose it's better to allow it to crash if its empty for whatever reason